airbyte-server/src/main/java/io/airbyte/server/handlers/ConnectionsHandler.java

```diff
@@ -268,6 +274,56 @@ public static CatalogDiff getDiff(final AirbyteCatalog oldCatalog, final Airbyte
         .toList());
   }
 
+  /**
+   * Returns the list of the streamDescriptor that have their config updated.
+   * @param oldCatalog the old catalog
+   * @param newCatalog the new catalog
+   * @return the list of StreamDescriptor that have their configuration changed
+   */
+  public List<StreamDescriptor> getConfigurationDiff(final AirbyteCatalog oldCatalog, final AirbyteCatalog newCatalog) {
+    final Map<StreamDescriptor, AirbyteStreamConfiguration> oldStreams = catalogToPerStreamConfiguration(oldCatalog);
+    final Map<StreamDescriptor, AirbyteStreamConfiguration> newStreams = catalogToPerStreamConfiguration(newCatalog);
+
+    final List<StreamDescriptor> streamWithDifferentConf = new ArrayList<>();
+
+    newStreams.forEach(((streamDescriptor, airbyteStreamConfiguration) -> {
+      final AirbyteStreamConfiguration oldConfig = oldStreams.get(streamDescriptor);
+
+      if (oldConfig == null) {
+        // The stream is a new one, the config has not change and it needs to be in the schema change list.
+      } else {
+        if (haveConfigChange(oldConfig, airbyteStreamConfiguration)) {
+          streamWithDifferentConf.add(streamDescriptor);
+        }
+      }
+    }));
+
+    return streamWithDifferentConf;
+  }
+
+  private boolean haveConfigChange(final AirbyteStreamConfiguration oldConfig, final AirbyteStreamConfiguration newConfig) {
```

Comment:
I don't think this method is actually necessary. You should just be able to do `oldConfig.equals(newConfig)` or `Objects.equals(oldConfig, newConfig)`, because the `AirbyteStreamConfiguration` generated class has a generated `equals()` method that is basically doing the same thing you did here. (I just tried that out locally and it worked)

And the benefit of using the generated equals method is that it is guaranteed to stay up to date with any changes to that class

Reply:
I didn't use it because I don't think that we need to compare all the fields. The order of the cursor and the primary key shouldn't have any impact on the equals here, that's why I have to convert them to set to calculate the diff (it could have been done by ordering as well).

It is important to not have all the fields because of the remove stream that we don't want in the list of stream being reset.

Reply:
@benmoriceau I just checked our documentation, and it looks like the order of the cursor field values _does_ matter. They are used to establish a path to the key that one wants to use as the cursor in the case of nested fields: https://docs.airbyte.com/understanding-airbyte/airbyte-protocol#logic-for-resolving-the-cursor-field

Relevant snippet from that page:
> If a configuration requires a cursor field, it requires an array of strings that serves as a path to the desired field. e.g. if the structure of a stream is `{ value: 2, metadata: { updated_at: 2020-11-01 } }` the default_cursor_field might be `["metadata", "updated_at"]`.

Reply:
Thanks, I guess that for the PK, the state of the first list won't matter but the state of the second list will.

I still need to keep it in a dedicated method because the `selected` property should not be part of  diff calculation.